dash/_validate.py

```diff
@@ -422,6 +423,15 @@ def validate_layout(layout, layout_value):
     component_ids = set()
 
     def _validate(value):
+        def _validate_type(comp):
+            deprecated_types = ["LogoutButton"]
```

Comment:
Just to make sure there is no collision, we should also check the `_namespace` attribute. Also could add a custom deprecation message to the component.

I would refactor the `deprecated_types` to a `collections.defaultdict(lambda: collections.defaultdict(dict))` then set:
```
deprecated_components["dash_core_components"]["LogoutButton"] = "''
The Logout Button is no longer used with Dash Enterprise and can be replaced with a html.Button or html.A.
eg: html.A(href=os.getenv('DASH_LOGOUT_URL'))
"""
```
The for the check you could have:

```
_type = getattr(comp, "_type",  "")
_ns = getattr(comp, "_namespace", "")
deprecation_message = deprecated_components[_ns][_type]
if deprecation_message:
    ...
```

Reply:
Changed in 87385a9